### PR TITLE
Issue #15: Set default config for S3 bucket

### LINKED ISSUE #15: The HeadBucket operation requires non-empty parameter: Bucket
I installed this module and the AWS module but I get the following error when opening the "Plugin Administration"

[WAR] d5 (lib/errors.php:529) [InvalidArgumentException]: The HeadBucket operation requires non-empty parameter: Bucket at /var/www/html/module/aws/sdk/Aws/InputValidationMiddleware.php:64

Is there extra configuration required other than what is mentioned in the README?

### PR DESCRIPTION
**Pull request by petersistrom**
Resolves #15

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -244,6 +244,7 @@ public static function save_config_options(Pieform $form, $values) {
     }
 
     public static function postinst($fromversion) {
```
Comment (matthewhilton) [inline lib.php:246]: Not 100% sure, but is this called when you upgrade the plugin? Or only on install?

If its called on upgrade we need a check for if the config already exists so we dont overwrite anything :smile:

Comment (petersistrom) [reply lib.php:246]: It's called on both install and upgrade. To call only on install you can add an `if ($fromversion == 0)` 
However, I think an empty check is simpler to handle both install and upgrade.

Comment (matthewhilton) [reply lib.php:246]: So if someone already has the plugin installed, then wouldn't this override their config on every upgrade ? Or am I missing something

Comment (matthewhilton) [reply lib.php:246]: Ah NVM i see you added a check

**Pull request merged**